modules/rest-http/src/main/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/GridJettyRestProtocol.java

```diff
@@ -129,7 +129,8 @@ public GridJettyRestProtocol(GridKernalContext ctx) {
             throw new IgniteCheckedException("Failed to resolve local host to bind address: " + ctx.config().getLocalHost(), e);
         }
 
-        System.setProperty(IGNITE_JETTY_HOST, locHost.getHostAddress());
+        if (System.getProperty(IGNITE_JETTY_HOST) == null)
```

Comment:
How about to move this "if" before locHost check?
In case we have IGNITE_JETTY_HOST we should try to resolve it instead of ctx.config().getLocalHost() ?

Reply:
Guys, we don't review changes at GitHub. Could you please create Upsource review?

Reply:
@agura As far as I remember the discussion, upsource is optional for such small fixes ;) Also it is specified in wiki https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute
Anyway, @akuznetsov-gridgain I go back to jira for more comments.